app/bin/tools_common.dart

```diff
@@ -6,34 +6,22 @@ import 'dart:io';
 import 'dart:async';
 
 import 'package:appengine/appengine.dart';
-import 'package:gcloud/service_scope.dart';
 import 'package:pub_dartlang_org/upload_signer_service.dart';
 
 import 'server_common.dart';
 import 'configuration.dart';
 
 Future withProdServices(Future fun()) {
-  if (Platform.isMacOS) {
-    return _withProdServices(() {
-      return fork(() async {
-        await initializeApiaryDatastore();
-        return fun();
-      });
+  return withCorrectDatastore(() {
+    if (!Platform.environment.containsKey('GCLOUD_PROJECT') ||
+        !Platform.environment.containsKey('GCLOUD_KEY')) {
+      throw 'Missing GCLOUD_* environments for package:appengine';
+    }
+    return withAppEngineServices(() {
```

Comment:
The `withCorrectDatastore` and `withAppEngineServices` need to be switched (maybe I got that wrong in my example).